### PR TITLE
06 core get /api/articles/article id/comments

### PR DESCRIPTION
**Pull request by RKHarrison**
Hi NC Team!

I added the /api/articles/:article_id/comments endpoint:
- added 200 happy path functionality and testing
- added 200 empty array response for when an article exists but has no linked comments
- added a custom 404 Resource Not Found error for when a request is made for an article id that is valid but non-existent
- handles invalid article id request with a 400 failing schema response
- updates endpoints.json accordingly
- removes unwanted working query files as per your previous comments
- updates .gitignore to reflect this

### CODE REVIEW

File: __tests__/app.test.js [hunk 1]
```diff
@@ -89,19 +90,72 @@ describe("GET /api/articles/:article_id", () => {
...
+    return request(app)
+      .get("/api/articles/2/comments")
+      .expect(200)
+      .then(({ body }) => {
+        const commentsForArticleId = body.commentsForArticleId;
+        expect(commentsForArticleId).toHaveLength(0);
+        expect(commentsForArticleId).toBeInstanceOf(Array);
+      });
+  });
+  it("404: responds 'Not Found' when given valid but non-existing article id", () => {
```
Comment (nc-poppy-m) [inline __tests__/app.test.js:142]: Lovely! Good distinction between 404 and 400

File: __tests__/app.test.js [hunk 2]
```diff
@@ -89,19 +90,72 @@ describe("GET /api/articles/:article_id", () => {
...
+
+describe("GET /api/articles/:article_id/comments", () => {
+  it("200: responds with an array of comments for the given article id, sorted by most recent, with the correct properties ", () => {
+    return request(app)
+      .get("/api/articles/1/comments")
+      .expect(200)
+      .then(({ body }) => {
+        const commentsForArticleId = body.commentsForArticleId;
+        expect(commentsForArticleId).toHaveLength(11);
+        expect(commentsForArticleId).toBeSortedBy("created_at", {
```
Comment (nc-poppy-m) [inline __tests__/app.test.js:117]: Nice use of jest sorted

File: api/controllers/app.api.comments.controllers.js [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+const {
+  fetchCommentsByArticleId,
+} = require("../models/app.api.comments.models");
+
+exports.getCommentsByArticleId = (req, res, next) => {
+  const { article_id } = req.params;
+
+  checkArticleExists(article_id)
+    .then(() => {
+      fetchCommentsByArticleId(article_id).then((commentsForArticleId) => {
```
Comment (nc-poppy-m) [inline api/controllers/app.api.comments.controllers.js:11]: This is nested. Make sure you change this, as you can't have a `.then()` block within another `.then()` block.

File: api/models/app.api.articles.models.js [hunk 4]
```diff
@@ -41,4 +23,36 @@ ORDER BY a.created_at DESC;
...
+        const article = rows[0]
+
+        if(!article) {
+            return Promise.reject({status: 404, msg: 'Resource Not Found'})
+        }
+        return article
+    })
+};
+
+exports.checkArticleExists = (article_id) => {
```
Comment (nc-poppy-m) [inline api/models/app.api.articles.models.js:46]: Good work!

**Review by nc-poppy-m — COMMENTED** [review]
Great work here! However, you've nested your `.then()` block in your controller. You don't need to resubmit the PR for this, but make sure you change that as soon as you can

**Comment by RKHarrison** [comment]
thanks so much Poppy! I'll unnest that pesky then block asap!

**Pull request merged**